File: DSA/Aarohi-Sharma/Day1_q1.cpp
#include<iostream>
using namespace std;
int main(){
    int n;
    cin>>n;
    int array[n];
    cout<<"Enter elements:";
    for (int i=0; i<n; i++){
        cin>>array[i];
    }
    for (int i=0; i<n; i++){
        cout<<array[i]<<endl;
    }
    int d, m;
    cin>>d>>m;
    int sum=0, count=0;
    for(int i=0; i<n; i++){
        for(int j=i+1; j<=i+m-1; j++){
            sum=array[i]+array[j];
            if(sum==d){
                count++;
            }
        }
    }
    cout<<count;
}

File: DSA/Ishita-Nigam/Day1/Day1_Q1.java
import java.util.Scanner;

public class Day1_Q1 {
    public static int countWays(int[] candy, int day, int month) {
        int count = 0;
        for (int i = 0; i <= candy.length - month; i++) {
            int sum = 0;
            for (int j = 0; j < month; j++) {
                sum += candy[i + j];
            }
            if (sum == day) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter array size :");
        int n = in.nextInt();
        int candy[] = new int[n];
        System.out.println("Enter array elements:");
        for (int i = 0; i < n; i++) {
            candy[i] = in.nextInt();
        }
        System.out.println("Enter birth day");
        int day = in.nextInt();
        System.out.println("Enter birth  month");
        int month = in.nextInt();
        System.out.println("Resultant is:" + countWays(candy, day, month));
        in.close();
    }
}